File: verwerken/plooi-dcn/dcn-admin-service/src/main/java/nl/overheid/koop/plooi/dcn/admin/model/BulkActionRequest.java
package nl.overheid.koop.plooi.dcn.admin.model;

import nl.overheid.koop.plooi.dcn.model.TriggerType;
import nl.overheid.koop.plooi.dcn.solr.models.SolrSearchRequest;

public record BulkActionRequest(TriggerType triggerType, SolrSearchRequest filter, String reason, String action) {

    public String triggerTypeName() {
        return this.triggerType.toString();
    }
}

File: verwerken/plooi-dcn/dcn-admin-client/src/app/process/models/bulk-action-request.ts
import {SolrSearchFilter} from "./solr-search-filter";

export class BulkActionRequest {
  triggerType: string;
  filter: SolrSearchFilter
  action : string;
  reason : string;

    constructor(triggerType: string, filter: SolrSearchFilter, action: string, reason: string) {
        this.triggerType = triggerType;
        this.filter = filter;
        this.action = action;
        this.reason = reason;
    }
}
